### ISSUE TITLE
Generated project_root symlink throws error in Intellij

### ISSUE THREAD

**Issue by naveenOnarayanan** [facebook/buck#2076]
### Background
With the recent changes to support buck-cells the following commit: https://github.com/facebook/buck/commit/8960005e2ec8900e8fc7c26a7b7239a7f512931a#diff-5ba521442393ce8db30deb6aa188a0eaR213 introduced a changeset that generates a symlink under `buck-out/project_root` that points back to the main project when running any sort of `buck build` command. 

This introduces a potential infinite recursion in Intellij as the `Git` plugin attempts to resolve all the git roots and ultimately drills deep down into the symlink infinitely which causes performance issues in Intellij.

```
VCS Root: [Git] -
2018-09-12 11:24:25,281 [  16921]   INFO - pl.projectlevelman.NewMappings - VCS Root: [Git] - [../buck-out/project_root] 
2018-09-12 11:24:25,281 [  16921]   INFO - pl.projectlevelman.NewMappings - VCS Root: [Git] - [../buck-out/project_root/buck-out/project_root] 
2018-09-12 11:24:25,281 [  16921]   INFO - pl.projectlevelman.NewMappings - VCS Root: [Git] - [../buck-out/project_root/buck-out/project_root/buck-out/project_root] 
2018-09-12 11:24:25,281 [  16921]   INFO - pl.projectlevelman.NewMappings - VCS Root: [Git] - [../buck-out/project_root/buck-out/project_root/buck-out/project_root/buck-out/project_root] 
2018-09-12 11:24:25,281 [  16921]   INFO - pl.projectlevelman.NewMappings - VCS Root: [Git] - [../buck-out/project_root/buck-out/project_root/buck-out/project_root/buck-out/project_root/buck-out/project_root] 
2018-09-12 11:24:25,281 [  16921]   INFO - pl.projectlevelman.NewMappings - VCS Root: [Git] - [../buck-out/project_root/buck-out/project_root/buck-out/project_root/buck-out/project_root/buck-out/project_root/buck-out/project_root]
```

**Comment by bobyangyf** [comment]
Do you have buck-out in your .gitignore? the git plugin shouldn't be looking into ignored dirs.

We'll be pushing out buck workspace soon and that should resolve this fully.

**Comment by artem-zinnatullin** [comment]
Yea, this definitely breaks IntelliJ

We have `buck-out` in `.gitignore`, adding a rule to exclude `buck-out/project_root` explicitly doesn't help either.

I'm not 100% sure that problem is caused by Git plugin in IntelliJ, it might be just indexing (this is when it gets stuck for us).

IntelliJ log is not very helpful, but here how it looks like:

```console
2018-10-29 13:22:50,944 [3904356]   INFO - .openapi.util.LowMemoryWatcher - Low memory signal received: afterGc=false
2018-10-29 13:22:58,092 [3911504]   INFO - .openapi.util.LowMemoryWatcher - Low memory signal received: afterGc=true
2018-10-29 13:23:05,619 [3919031]   INFO - .openapi.util.LowMemoryWatcher - Low memory signal received: afterGc=true
2018-10-29 13:23:12,727 [3926139]   INFO - .openapi.util.LowMemoryWatcher - Low memory signal received: afterGc=true
2018-10-29 13:23:19,672 [3933084]   INFO - .openapi.util.LowMemoryWatcher - Low memory signal received: afterGc=true
2018-10-29 13:23:27,109 [3940521]   INFO - .openapi.util.LowMemoryWatcher - Low memory signal received: afterGc=true
```

**Comment by artem-zinnatullin** [comment]
@bobyangyf what if instead of a symlink it would be a file with the path to the project root as text? 🤔

Symlink to parent directory can break a lot of tools and is pretty dangerous imho, at the same time, I'd expect tools to deal with that if they want to follow symlinks

**Comment by artem-zinnatullin** [comment]
Yeah, it's certainly Git plugin, it spends ridiculous amount of time scanning roots

(profiling when I just opened project in IntelliJ 2018.2.5 and did nothing for few mins)

<img width="1792" alt="screen shot 2018-10-29 at 4 14 46 pm" src="https://user-images.githubusercontent.com/967132/47685938-d8e07980-db95-11e8-9ce4-634cdb60db2d.png">

**Comment by artem-zinnatullin** [comment]
Reported to IntelliJ tracker https://youtrack.jetbrains.com/issue/IDEA-201436

I still think it's better to avoid such a symlink in the first place

**Comment by kageiit** [comment]
@styurin @mkillianey @bobyangyf Can we stop creating this symlink?

**Comment by bobyangyf** [comment]
Not right now. I will look into changing the buck directory structure to possibly fix this. Or put it behind a flag.

**Comment by artem-zinnatullin** [comment]
What about replacing symlink with plain text file containing required path?

**Comment by bobyangyf** [comment]
Sure I can do that as a hack for now.

**Comment by artem-zinnatullin** [comment]
Perfect, thanks @bobyangyf 

Fix was landed to master, just updating the issue: https://github.com/facebook/buck/commit/4bc92bf1049649dcb4fbe643825146fba463b3bb

**Comment by mikeb01** [comment]
Any idea when a version with this fix will be released?
